x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/OptimizerRules.java

```diff
@@ -105,6 +106,10 @@ protected AttributeSet references(LogicalPlan plan) {
                 // But they are not actually referring to attributes from the input plan - only the match field does.
                 return enrich.matchField().references();
             }
+            if (plan instanceof Lookup lookup) {
+                // NOCOMMIT what in the world am I doing?
+                return Expressions.references(lookup.matchFields());
```

Comment:
I have no idea why I had to do this. It feels wrong.

Reply:
Likely because the match fields are not found in the output - this is indeed a red flag that we should look into.